datadog_checks_base/datadog_checks/base/utils/db/statement_metrics.py

```diff
@@ -104,6 +113,47 @@ def compute_derivative_rows(self, rows, metrics, key):
         return result
 
 
+def merge_duplicate_rows(rows, metrics, key):
+    """
+    Given a list of query rows, apply query normalization and compute query signatures to detect duplicate
+    queries_by_key and merge them into a single row.
+
+    - **rows** (_List[dict]_) - rows from current check run
+    - **metrics** (_List[str]_) - the metrics to compute for each row
+    - **key** (_callable_) - function for an ID which uniquely identifies a query row across runs
+    """
+
+    queries_by_key = OrderedDict()
```

Comment:
Wondering what others think about this. Because python2 doesn't offer insertion ordering guarantees when returning values, I made this an `OrderedDict`. I don't know if it would actually matter much in practice if the rows were returned in a different order and it seems to me like it shouldn't matter and nothing downstream should assume this ordering. 

That said, the tests are assuming this ordering so if we'd prefer just using the default `dict`, I need to make changes to at least one test to accommodate for that.

Reply:
Because of the nature of merging, I don't think ordering is a necessary guarantee. Because I may expect my rows to be ordered by "count desc" but after merging the bottom 10 rows together, they should actually be at the top. A plain old dict should be fine here.